snuba_sdk/metrics_visitors.py

```diff
@@ -212,14 +214,17 @@ def visit(self, metric: Metric) -> TVisited:
         raise NotImplementedError
 
 
-class MetricMQLPrinter(MetricVisitor[Mapping[str, Union[str, Mapping[str, str]]]]):
-    def visit(self, metric: Metric) -> Mapping[str, str | Mapping[str, str]]:
+class MetricMQLPrinter(
+    MetricVisitor[Mapping[str, Union[str, Mapping[str, Union[str, None]]]]]
+):
+    def visit(
+        self, metric: Metric
+    ) -> Mapping[str, str | Mapping[str, Union[str, None]]]:
```

Comment:
If we could make this type more descriptive that would be cool. Right now it's complex and I have a hard time understanding what this function is actually returning

Reply:
Honestly the typing of these functions needs to be reworked in general. These functions are tied to the base functions of the generic class, and that has been a problem in general with metrics, since we want to be able to return more complex types and not be locked into the generic type defined for the class.